<?xml version="1.0"?>
<launch>

	<!--<include file="$(find zed_wrapper)/launch/zedm.launch"/>  -->

	<node pkg="rosbag" type="play" name="player" output="log" args="--clock /home/jlew/bagfiles/uc/modex_full2.bag"/>

	<node pkg="web_video_server" name="video_server" type="web_video_server" respawn="true" respawn_delay="5">
          <param name="port" value="8080"/>
          <param name="server_threads" value="2"/>
          <param name="ros_threads" value="2"/>
	</node>

    <node pkg="octomap_server" type="modex_rack.py" name="check_rack_node" output="screen" />

	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
		<param name="resolution" value="0.05" />
		
		<!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
		<param name="frame_id" type="string" value="map" />
		
		<!-- maximum range to integrate (speedup!) -->
		<param name="sensor_model/max_range" value="2.5" />
       
	    <param name="pointcloud_min_z" value="0.1"/>
	    <param name="pointcloud_max_z" value="3.4"/>

        <param name="sensor_model/min" value="0.12"/>
        <param name="sensor_model/max" value="0.97"/>

        <param name="sensor_model/hit"   value="0.85" />
        <param name="sensor_model/miss"   value="0.25" />    
	
		<!-- data source to integrate (PointCloud2) -->
		<remap from="cloud_in" to="/zed1/point_cloud/cloud_registered" />
	</node>

	<node pkg="zed-tensorflow" name="object_detection" type="object_detection_modex.py" />
    <!-- Launch rviz display-->
	<node name="rviz" pkg="rviz" type="rviz" args="-d /home/jlew/octo.rviz" output="screen" />

</launch>